src/pudl/transform/epacems.py

```diff
@@ -42,160 +46,243 @@ def harmonize_eia_epa_orispl(
+        core_eia__entity_plants: EIA plants DataFrame.
 
     Returns:
         A single year_quarter of EPA CEMS data
     """
-    # Create all the table inputs used for the subtransform functions below
-
     return (
-        raw_df.pipe(apply_pudl_dtypes, group="epacems")
-        .pipe(remove_leading_zeros_from_numeric_strings, "emissions_unit_id_epa")
-        .pipe(harmonize_eia_epa_orispl, core_epa__assn_eia_epacamd)
-        .pipe(
-            convert_to_utc,
-            plant_utc_offset=_load_plant_utc_offset(core_eia__entity_plants),
+        raw_lf.pipe(apply_pudl_dtypes_polars, group="epacems")
+        .with_columns(
+            # Strip leading zeros from strings
+            # TODO: Update method in helpers.py with polars implementation from here.
+            emissions_unit_id_epa=pl.when(
+                pl.col("emissions_unit_id_epa").str.contains(r"^\d+$")
+            )
+            .then(pl.col("emissions_unit_id_epa").str.strip_chars_start("0"))
+            .otherwise(pl.col("emissions_unit_id_epa"))
+        )
+        .pipe(harmonize_eia_epa_orispl, core_epa__assn_eia_epacamd.lazy())
+        .pipe(convert_to_utc, plant_utc_offset=plant_utc_offset.lazy())
+        # Fix gross load that is orders of magnitude too high
+        .with_columns(
+            gross_load_mw=pl.when(pl.col("gross_load_mw") > 2000)
+            .then(pl.col("gross_load_mw") / 1000)
+            .otherwise(pl.col("gross_load_mw"))
         )
-        .pipe(correct_gross_load_mw)
-        .pipe(apply_pudl_dtypes, group="epacems")
+        .pipe(apply_pudl_dtypes_polars, group="epacems")
     )
+
+
+def _partitioned_path() -> Path:
+    partitioned_path = (
+        PudlPaths().output_dir / "parquet" / "raw_epacems__hourly_emissions"
+    )
+    partitioned_path.mkdir(exist_ok=True)
+    return partitioned_path
+
+
+@dg.asset(
+    required_resource_keys={"datastore", "dataset_settings"},
+    io_manager_key="parquet_io_manager",
+)
+def core_epacems__hourly_emissions(
+    context,
+    _core_epa__assn_eia_epacamd_unique: pd.DataFrame,
+    core_eia__entity_plants: pd.DataFrame,
+) -> pl.LazyFrame:
+    """Extract EPA CEMS data by quarter and write to partitioned Parquet files."""
+    # Validate uniqueness once before processing all partitions
+    unique_crosswalk = pl.DataFrame(_core_epa__assn_eia_epacamd_unique)
+    _validate_crosswalk_uniqueness(unique_crosswalk)
+    plant_utc_offset = _load_plant_utc_offset(pl.DataFrame(core_eia__entity_plants))
+    partitioned_path = _partitioned_path()
+
+    # Iterate over all the partitions we are processing, since polars is parallelized
+    # internally and this will save us significant dagster process startup overhead and
+    # avoid CPU resource contention.
+    output_paths = []
+    for year_quarter in context.resources.dataset_settings.epacems.year_quarters:
+        output_path = partitioned_path / f"{year_quarter}.parquet"
+        context.log.info(f"Processing EPA CEMS {year_quarter}")
```

Comment:
I had not come across `context.log` before but it integrates the logging output directly into Dagster which was nice.